reference/api/README.md

```diff
@@ -14,9 +14,13 @@ Check out [the FAQ](/resources/faq.md) for answers to some common questions 💡
 
 #### Content type
 
-Every request and response body is in `JSON`.
+ Requests can be in JSON, CSV, or NDJSON. Responses are always in JSON.
 
-It is **not required** to have `Content-Type: application/json` in the header. Any content-type is accepted.
+You **must** specify a Content-Type header. MeiliSearch currently supports the following formats:
+
+- `Content-Type: application/json` for JSON
+- `Content-Type: application/x-ndjson` for NDJSON
+- `Content-Type: text/csv` for CSV
```

Comment:
CSV and NDJSON are only supported for `/indexes/:uid/documents` route (`POST` and `PUT`).
You should not be able to push your setting in CSV via the `/indexes/:uid/settings` routes that only accept JSON.
@gmourier can you confirm?

Reply:
Yes, I confirm!

Reply:
It is mandatory to specify `Content-Type` headers for every `PUT` and `POST` routes, correct?

Reply:
Nope, only if we pass a payload. For example, the route to create a dump is `POST` but does not require any payload so no need to enforce the content type. I think this is the only exception, but I might be wrong.

Reply:
Got it, thanks, @curquiza!

Reply:
updated